src/twodturb.jl

```diff
@@ -100,50 +96,70 @@ end
 # Vars
 # ----
 
-varspecs = cat(
-  getfieldspecs(physicalvars, :(Array{T,2})),
-  getfieldspecs(transformvars, :(Array{Complex{T},2})),
-  dims=1)
+struct Vars{Aphys, Atrans} <: TwoDTurbVars
+   zeta :: Aphys
+      u :: Aphys
+      v :: Aphys
+  zetah :: Atrans
+     uh :: Atrans
+     vh :: Atrans
+end
 
-forcedvarspecs = cat(varspecs, getfieldspecs(forcedvars, :(Array{Complex{T},2})), dims=1)
-stochforcedvarspecs = cat(forcedvarspecs, getfieldspecs(stochforcedvars, :(Array{Complex{T},2})), dims=1)
+struct ForcedVars{Aphys, Atrans} <: TwoDTurbVars
+   zeta :: Aphys
+      u :: Aphys
+      v :: Aphys
+  zetah :: Atrans
+     uh :: Atrans
+     vh :: Atrans
+     Fh :: Atrans
+end
+
+struct StochasticForcedVars{Aphys, Atrans} <: TwoDTurbVars
+     zeta :: Aphys
+        u :: Aphys
+        v :: Aphys
+    zetah :: Atrans
+       uh :: Atrans
+       vh :: Atrans
+       Fh :: Atrans
+  prevsol :: Atrans
+end
```

Comment:
ya!

We could maybe add special type parameters for `Fh` and `prevsol` and just have one type for all the vars. When both are `Nothing`, we have an ordinary twodturb problem. When just `prevsol` is nothing, it's a forced problem. Otherwise, its a stochastic forced problem...

Reply:
If you do that you can keep the constructor names. Just need to make sure they do the *right thing*.

Reply:
Can you try do that? I've lost my patience with `@eval`...